src/Service/UserManager.php

```diff
@@ -308,30 +311,35 @@ public function detachCover(User $user): void
         $this->bus->dispatch(new DeleteImageMessage($image));
     }
 
-    public function deleteRequest(User $user): void
+    public function deleteRequest(User $user, bool $immediately): void
     {
-        $user->markedForDeletionAt = new \DateTime();
-        $user->visibility = VisibilityInterface::VISIBILITY_SOFT_DELETED;
+        if (!$immediately) {
+            $user->markedForDeletionAt = date_add(new \DateTime(), new \DateInterval('P30D'));
+            $user->isDeleted = true;
 
-        $this->entityManager->persist($user);
-        $this->entityManager->flush();
+            $this->entityManager->persist($user);
+            $this->entityManager->flush();
+        } else {
+            $this->delete($user);
+        }
     }
 
-    public function revokeDeleteRequest(User $user): void
+    public function removeDeleteRequest(User $user): void
```

Comment:
Add code documentation above these methods, so other developers do not need to read all the lines in the future to understand what each of these methods do. 

Also just in general this can help onboard new developers to the project.